reps/2023-04-28-remove-algorithms-from-rllib.md

```diff
@@ -0,0 +1,193 @@
+## Summary
+
+We'd like to remove aprox ~20 of the 30 algorithms from RLlib. We are considering doing this because 
+
+1. It would reduce the maintenance burden of RLlib.
+2. Many of these algorithms have successors that are more performant and/or easier to hyperparameter tune.
+3. The proposed algorithms have low ussage by the community according to our ray cluster telemetry readings.
+
+However, we recognize that the value that these algorithms have to the community, and so we are establishing the
+`rayproject/rllib-contrib` repo to house these algorithms. Each deprecated algorithm will have its own directory
+in this repo containing the following:
+
+1. The implementation of the algorithm.
+2. Any tests that are associated with ensuring the correctness of the algorithm.
+3. End to end example scripts that demonstrate how to run the algorithm
+4. The necessary requirements for running the algorithm
+5. boiler plate that allows the algorithm to be installed as a pip package.
+
+
+## Ussage
+
+Users will be able to replace the deprecated algorithms by doing the following:
+
+1. use git to clone `rayproject/rllib-contrib`
+2. navigate to the directory of the previously deprecated algorithm
+   for ex. `cd rllib-contrib/maml`
+3. run `pip install -e .` to install the algorithm as a pip package
+4. Import the algorithm, its policies and its config from that package instead of from `ray.rllib`
+
+In most usecases, the migration will look like replacing imports inside of an experiment script.
+
+For example:
+
+```python
+from gymnasium.wrappers import TimeLimit
+
+import ray
+from ray import air
+from ray import tune
+from ray.rllib.examples.env.cartpole_mass import CartPoleMassEnv
+from ray.tune.registry import register_env
+
+# from ray.rllib.algorithms.maml import MAML, MAMLConfig  # BEFORE
+from rllib_maml.maml import MAML, MAMLConfig  # AFTER
+
+if __name__ == "__main__":
+    ray.init()
+    register_env(
+        "cartpole",
+        lambda env_cfg: TimeLimit(CartPoleMassEnv(), max_episode_steps=200),
+    )
+
+    config = MAMLConfig()
+
+    tuner = tune.Tuner(
+        MAML,
+        param_space=config.to_dict(),
+        run_config=air.RunConfig(stop={"training_iteration": 100}),
+    )
+    results = tuner.fit()
+
+```
+
+
+This proposed algorithms to remove are:
+- A3C
+- AphaStar
+- AlphaZero
+- ApexDQN
```

Comment:
Why keep DQN but remove ApexDQN?

Reply:
@avnishn Because it's such a foundational algorithm? But we are keeping PG? I don't understand either.

Reply:
DQN is easy to tune for off policy rl discrete action spaces compared to all the other off policy RL algorithms. 

That being said, APEX-DQN has slightly less performance than APPO / impala. So with that being said it just doesn't make sense to keep since users could just use APPO.

Reply:
Actually, we are not keeping PG either (and we shouldn't).

Reply:
also DQN is the third most used algorithm in RLlib oss.